src/video/allegro_v.cpp

```diff
@@ -464,9 +463,9 @@ void VideoDriver_Allegro::InputLoop()
 	if (key[KEY_TAB] && (key_shifts & KB_ALT_FLAG) == 0)
 #endif
 	{
-		if (!_networking && _game_mode != GM_MENU) _fast_forward |= 2;
-	} else if (_fast_forward & 2) {
-		_fast_forward = 0;
+		this->fast_forward_key_pressed = true;
+	} else {
+		this->fast_forward_key_pressed = false;
```

Comment:
Stick this (and the other one) in the #if blocks, perhaps?
`this->fast_forward_key_press = _shift_pressed`
etc

Alternatively, dare I ask if it's possible to factor out of the specific video-driver code entirely?